ros2_batch_job/__main__.py

```diff
@@ -358,13 +357,20 @@ def build_and_test(args, job):
         return ret_build
 
     print('# BEGIN SUBSECTION: test')
-    ret_test = job.run([
+
+    test_cmd = [
         args.colcon_script, 'test',
         '--base-paths', '"%s"' % args.sourcespace,
         '--build-base', '"%s"' % args.buildspace,
         '--install-base', '"%s"' % args.installspace,
-    ] + (['--merge-install'] if not args.isolated else []) + args.test_args,
-        exit_on_error=False, shell=True)
+    ]
+    if not args.isolated:
+        test_cmd.append('--merge-install')
+    if args.coverage:
+        test_cmd.append('--pytest-with-coverage')
+    test_cmd.append(args.test_args)
```

Comment:
This should be `test_cmd += args.test_args` instead.

Reply:
Ups, done in https://github.com/ros2/ci/pull/373/commits/04cf53f52ac97936b09a2caa6f866ec44bda7aa1.